CMakeLists.txt

```diff
@@ -89,9 +89,17 @@ find_path(LIBMAD_INCLUDE_DIR mad.h)
 find_library(LIBMAD_LIB_DEBUG NAMES mad libmad NAMES_PER_DIR PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib")
 find_library(LIBMAD_LIB_RELEASE NAMES mad libmad NAMES_PER_DIR PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib")
 
-# FLAC's config is missing find_dependency(Threads), so include it here.
-find_package(Threads REQUIRED)
-find_package(FLAC CONFIG REQUIRED)
+if(UNIX AND NOT APPLE AND NOT ES_USE_VCPKG)
+	# Many linux distros use the autotools build of flac, which does not
+	# package the flac-config.cmake. Use PkgConfig instead.
+	find_package(PkgConfig REQUIRED)
+	pkg_check_modules(flac++ REQUIRED IMPORTED_TARGET flac++)
+	add_library(FLAC::FLAC++ ALIAS PkgConfig::flac++)
```

Comment:
This means adding PkgConfig as dependency. Can we make it so that PkgConfig is required only if the standard way fails?